From 60c376d1ca7029da5d5a636f89a6ae1d41ba00f4 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Guido=20G=C3=BCnther?= <agx@sigxcpu.org>
Date: Mon, 11 Mar 2024 10:45:17 +0100
Subject: [PATCH 1/7] background: Use getters
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Easier to read and avoids g_object_get() round trip

Signed-off-by: Guido Günther <agx@sigxcpu.org>
Part-of: <https://gitlab.gnome.org/World/Phosh/phosh/-/merge_requests/1383>
(cherry picked from commit a43767158b9ee4c5a8190bd75f35674057f925f1)
---
 src/background.c | 10 ++++++----
 1 file changed, 6 insertions(+), 4 deletions(-)

diff --git a/src/background.c b/src/background.c
index 00b61dbb7..fd22333a6 100644
--- a/src/background.c
+++ b/src/background.c
@@ -274,12 +274,14 @@ update_image (PhoshBackground *self)
 {
   int width, height;
 
-  if (self->primary)
+  if (self->primary) {
     phosh_shell_get_usable_area (phosh_shell_get_default (), NULL, NULL, &width, &height);
-  else
-    g_object_get (self, "configured-width", &width, "configured-height", &height, NULL);
+  } else {
+    width = phosh_layer_surface_get_configured_width (PHOSH_LAYER_SURFACE (self));
+    height = phosh_layer_surface_get_configured_height (PHOSH_LAYER_SURFACE (self));
+  }
 
-  g_debug ("Scaling %p to %dx%d", self, width, height);
+  g_debug ("Scaling background %p to %dx%d", self, width, height);
 
   g_clear_object (&self->pixbuf);
   self->pixbuf = image_background (self->cached_bg_image, width, height,
-- 
GitLab


From d10765d162492ffd605b390a4e61c4b424aba6c1 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Guido=20G=C3=BCnther?= <agx@sigxcpu.org>
Date: Mon, 11 Mar 2024 10:50:05 +0100
Subject: [PATCH 2/7] background-manager: Drop unused param from doc string
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Gbp-Dch: Ignore
Signed-off-by: Guido Günther <agx@sigxcpu.org>
Part-of: <https://gitlab.gnome.org/World/Phosh/phosh/-/merge_requests/1383>
(cherry picked from commit bd2087e1cb8b4b5639d8f0abdc4a6e57f863dc15)
---
 src/background-manager.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/background-manager.c b/src/background-manager.c
index 94588553d..153c9d98e 100644
--- a/src/background-manager.c
+++ b/src/background-manager.c
@@ -234,6 +234,7 @@ on_monitor_configured (PhoshBackgroundManager *self, PhoshMonitor *monitor)
   float scale;
 
   g_return_if_fail (PHOSH_IS_MONITOR (monitor));
+
   scale = phosh_monitor_get_fractional_scale (monitor);
   g_debug ("Monitor %p (%s) configured, scale %f", monitor, monitor->name, scale);
 
@@ -414,9 +415,8 @@ phosh_background_manager_get_backgrounds (PhoshBackgroundManager *self)
  * phosh_background_manager_get_data:
  * @self: The background manager
  * @background: The background to fetch information fore
- * @scale: The scale to use
  *
- * Get the data to that allows a [type@Background] to build it's
+ * Get the data that allows a [type@Background] to build it's
  * image. This is the single place that determines this so other parts
  * don't need to care whether we handle a slide or a single image.
  *
-- 
GitLab


From bddceb7a6a872046e6eaea996893d1694de41d6f Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Guido=20G=C3=BCnther?= <agx@sigxcpu.org>
Date: Mon, 11 Mar 2024 11:17:20 +0100
Subject: [PATCH 3/7] background-manager: Refresh the background on monitor
 configuration changes

Otherwise we keep using the old pixmap which likely doesn't fit the
current scale or screen orientation.

Closes: https://gitlab.gnome.org/World/Phosh/phosh/-/issues/1035
Part-of: <https://gitlab.gnome.org/World/Phosh/phosh/-/merge_requests/1383>
(cherry picked from commit ba43d796764c1e56873c045d45508bd5bf2ca1d5)
---
 src/background-manager.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/background-manager.c b/src/background-manager.c
index 153c9d98e..c9e2c3afc 100644
--- a/src/background-manager.c
+++ b/src/background-manager.c
@@ -242,6 +242,8 @@ on_monitor_configured (PhoshBackgroundManager *self, PhoshMonitor *monitor)
   if (background == NULL) {
     background = create_background_for_monitor (self, monitor);
     g_hash_table_insert (self->backgrounds, g_object_ref (monitor), background);
+  } else {
+    phosh_background_needs_update (background);
   }
 
   gtk_widget_show (GTK_WIDGET (background));
-- 
GitLab


From 5549ba458aca8c6599d1c579405082534f41c030 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Guido=20G=C3=BCnther?= <agx@sigxcpu.org>
Date: Mon, 11 Mar 2024 16:55:10 +0100
Subject: [PATCH 4/7] background: Exit early if layer surface isn't fully
 configured yet
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

We can't query width or height if configuration isn't done.

Signed-off-by: Guido Günther <agx@sigxcpu.org>
Part-of: <https://gitlab.gnome.org/World/Phosh/phosh/-/merge_requests/1383>
(cherry picked from commit 8c2b2adb1276f64f5f9cf4fded5dc215d4e85da9)
---
 src/background.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/src/background.c b/src/background.c
index fd22333a6..178ce225b 100644
--- a/src/background.c
+++ b/src/background.c
@@ -274,6 +274,9 @@ update_image (PhoshBackground *self)
 {
   int width, height;
 
+  if (!self->configured)
+    return;
+
   if (self->primary) {
     phosh_shell_get_usable_area (phosh_shell_get_default (), NULL, NULL, &width, &height);
   } else {
@@ -281,6 +284,8 @@ update_image (PhoshBackground *self)
     height = phosh_layer_surface_get_configured_height (PHOSH_LAYER_SURFACE (self));
   }
 
+  g_return_if_fail (width > 0 && height > 0);
+
   g_debug ("Scaling background %p to %dx%d", self, width, height);
 
   g_clear_object (&self->pixbuf);
-- 
GitLab


From c9d44d752ae8c76aba0f9908e6277d6c9ea2390b Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Guido=20G=C3=BCnther?= <agx@sigxcpu.org>
Date: Sun, 31 Mar 2024 12:28:04 +0200
Subject: [PATCH 5/7] session: Only append session manager if supported by
 gnome-session
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

gnome-session dropped support for --builtin and --systemd. So don't
append that argument if it doesn't show up in gnome-session's help
output.

Signed-off-by: Guido Günther <agx@sigxcpu.org>
---
 data/phosh-session.in | 12 +++++++++++-
 1 file changed, 11 insertions(+), 1 deletion(-)

diff --git a/data/phosh-session.in b/data/phosh-session.in
index 6c295ab80..8bac2fc31 100755
--- a/data/phosh-session.in
+++ b/data/phosh-session.in
@@ -50,8 +50,18 @@ if [ -f "$HOME/.phoshdebug" ]; then
   . "$HOME/.phoshdebug"
 fi
 
+SESSION_MANAGER=
+# old gnome-session: Pass --builtin or --systemd
+if gnome-session --help | grep -qs '\-\-builtin'; then
+  SESSION_MANAGER=@session_manager@
+# gnome-session >= 46 doesn't support builtin
+elif [ "@session_manager@" = "--builtin" ]; then
+  echo "ERR: gnome-session only supports systemd, session cannot start" 1>&2
+  exit 1
+fi
+
 # Run gnome-session through a login shell so it picks
 # variables from /etc/profile.d (XDG_*)
 [ -n "$WLR_BACKENDS" ] || WLR_BACKENDS=drm,libinput
 export WLR_BACKENDS
-exec "${COMPOSITOR}" -S -C "${PHOC_INI}" -E "bash -lc 'exec ${GNOME_SESSION} --disable-acceleration-check --session=phosh @session_manager@'"
+exec "${COMPOSITOR}" -S -C "${PHOC_INI}" -E "bash -lc 'exec ${GNOME_SESSION} --disable-acceleration-check --session=phosh ${SESSION_MANAGER}'"
-- 
GitLab


From 80957091329c7329e6064340814ac8d1009d6db9 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Guido=20G=C3=BCnther?= <agx@sigxcpu.org>
Date: Sun, 31 Mar 2024 12:29:30 +0200
Subject: [PATCH 6/7] build: Use systemd session manager by default
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

gnome-session dropped support for the builtin manager in 46.

Signed-off-by: Guido Günther <agx@sigxcpu.org>
---
 meson_options.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/meson_options.txt b/meson_options.txt
index 2320bd75f..940cf9b1c 100644
--- a/meson_options.txt
+++ b/meson_options.txt
@@ -15,7 +15,7 @@ option('man',
        description : 'generate man pages (requires rst2man)')
 
 option('systemd',
-       type: 'boolean', value: false,
+       type: 'boolean', value: true,
        description: 'Whether to generate systemd user units')
 
 option('compositor',
-- 
GitLab


From 7bc06aa60e87d1b653421c34a8cc0d6667a381fd Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Guido=20G=C3=BCnther?= <agx@sigxcpu.org>
Date: Thu, 21 Mar 2024 15:09:48 +0100
Subject: [PATCH 7/7] treewide: Document changes and release 0.37.1

---
 NEWS             |  7 +++++++
 debian/changelog | 22 ++++++++++++++++++++++
 meson.build      |  2 +-
 3 files changed, 30 insertions(+), 1 deletion(-)

diff --git a/NEWS b/NEWS
index 08800312e..e637bcfae 100644
--- a/NEWS
+++ b/NEWS
@@ -1,3 +1,10 @@
+phosh 0.37.1
+------------
+Released March 2024
+* Fix background scaling after rotation
+* Adjust session startup for gnome-session >= 46 which broke all
+  backwards compat by not accepting --systemd anymore.
+
 phosh 0.37.0
 ------------
 Released March 2024
diff --git a/debian/changelog b/debian/changelog
index be1ec9cea..f25ed0b58 100644
--- a/debian/changelog
+++ b/debian/changelog
@@ -1,3 +1,25 @@
+phosh (0.37.1) experimental; urgency=medium
+
+  * background: Use getters.
+    Easier to read and avoids g_object_get() round trip
+    (cherry picked from commit a43767158b9ee4c5a8190bd75f35674057f925f1)
+  * background-manager: Refresh the background on monitor configuration changes.
+    Otherwise we keep using the old pixmap which likely doesn't fit the
+    current scale or screen orientation.
+    Closes: https://gitlab.gnome.org/World/Phosh/phosh/-/issues/1035
+    (cherry picked from commit ba43d796764c1e56873c045d45508bd5bf2ca1d5)
+  * background: Exit early if layer surface isn't fully configured yet.
+    We can't query width or height if configuration isn't done.
+    (cherry picked from commit 8c2b2adb1276f64f5f9cf4fded5dc215d4e85da9)
+  * session: Only append session manager if supported by gnome-session
+    gnome-session dropped support for --builtin and --systemd. So don't
+    append that argument if it doesn't show up in gnome-session's help
+    output.
+  * build: Use systemd session manager by default
+    gnome-session dropped support for the builtin manager in 46.
+
+ -- Guido Günther <agx@sigxcpu.org>  Thu, 21 Mar 2024 15:08:39 +0100
+
 phosh (0.37.0) experimental; urgency=medium
 
   [ Arun Mani J ]
diff --git a/meson.build b/meson.build
index d300d39b1..bfeee5a16 100644
--- a/meson.build
+++ b/meson.build
@@ -1,5 +1,5 @@
 project('phosh', 'c',
-          version: '0.37.0',
+          version: '0.37.1',
           license: 'GPL-3.0-or-later',
     meson_version: '>= 1.0.0',
   default_options: [ 'warning_level=1', 'buildtype=debugoptimized', 'c_std=gnu11' ],
-- 
GitLab

